File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        return sorted(s) == sorted(t)


## previous logic
# convert srings to lists of chars and see if char[i] exists in char[j] 
# and continue unitl full lists are checked then true else false
# 
# sorted() creates a new list to store the characters; scales linearly

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {
        int[] in = new int[n + 1];

        for (int i = 1; i <= n; i++) {
            in[i] = in[i >> 1] + (i & 1);
        }
        return in;
    }
}
